### PR TITLE
[github] Check for 404 on /repositories pagination

### PR DESCRIPTION
**Pull request by pjlast**
The `/repositories` endpoint to fetch public repositories will erroneously indicate that there is a next page, even when there isn't one, and produce a 404 error when trying to fetch the next page. This fix addresses that scenario.

## Test plan

Tests should still pass

### CODE REVIEW

**Review by sashaostrikov — APPROVED** [review]
LGTM! Is it worth adding the test?

**Comment by pjlast** [comment]
@sashaostrikov maybe. These tests are always a pain in the butt to add 😩  I'll try

**Review by erzhtor — APPROVED** [review]

**Comment by sashaostrikov** [comment]
> These tests are always a pain in the butt to add 😩 I'll try

I totally agree, thanks for taking a stab, you're my hero 🙏🏻

File: internal/repos/github_test.go [hunk 1]
```diff
@@ -124,6 +124,63 @@ func TestGitHub_stripDateRange(t *testing.T) {
...
+
+	ctx := context.Background()
+	githubSrc, err := NewGitHubSource(ctx, logtest.Scoped(t), service, factory)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	results := make(chan *githubResult)
+	go func() {
+		githubSrc.listPublic(context.Background(), results)
```
Comment (sashaostrikov) [inline internal/repos/github_test.go:152]:
```suggestion
		githubSrc.listPublic(ctx, results)
```

File: internal/repos/github_test.go [hunk 2]
```diff
@@ -124,6 +124,63 @@ func TestGitHub_stripDateRange(t *testing.T) {
...
+	if count != 100 {
+		t.Errorf("unexpected repo count, wanted: 100, but got: %d", count)
+	}
+	if countArchived != 1 {
+		t.Errorf("unexpected archived repo count, wanted: 1, but got: %d", countArchived)
+	}
+}
+
+func prepareGheToken(t *testing.T, fixtureName string) string {
+	gheToken := os.Getenv("GHE_TOKEN")
```
Comment (sashaostrikov) [inline internal/repos/github_test.go:176]: probably we can add:
```suggestion
func prepareGheToken(t *testing.T, fixtureName string) string {
    t.Helper()
	gheToken := os.Getenv("GHE_TOKEN")
```

**Review by sashaostrikov — APPROVED** [review]
Thanks for adding tests! Left a couple of minor comments

**Comment by camdencheek** [comment]
Does this need backported to 5.0.4?

**Pull request merged**